src/services/node-bridge-data.provider.ts

```diff
@@ -1,72 +1,61 @@
-import {getMetricLogger} from '../utils/metric-logger';
-import {RskBridgeDataProvider} from './rsk-bridge-data.provider';
 import {RskNodeService} from './rsk-node.service';
-import { BlockTransactionObject } from 'web3-eth';
+import FilteredBridgeTransactionProcessor from './filtered-bridge-transaction-processor';
+import RskBlockProcessorPublisher from './rsk-block-processor-publisher';
 
-export class NodeBridgeDataProvider implements RskBridgeDataProvider {
+export class NodeBridgeDataProvider implements RskBlockProcessorPublisher {
   rskNodeService: RskNodeService;
   logger: Logger;
   filters: Array<BridgeDataFilterModel>;
+  private subscribers: FilteredBridgeTransactionProcessor[];
   constructor(
     @inject(ServicesBindings.RSK_NODE_SERVICE)
     rskNodeService: RskNodeService
   ) {
     this.rskNodeService = rskNodeService;
     this.filters = [];
     this.logger = getLogger('nodeBridgeDataProvider');
+    this.subscribers = [];
   }
-  async getData(startingBlock: string | number): Promise<BridgeData> {
-    const metricLogger = getMetricLogger(this.logger, 'getData');
-    const data: BridgeData = new BridgeData();
-    const lastBlock: BlockTransactionObject = await this.rskNodeService.getBlock(startingBlock);
-    if (lastBlock == null) {
-      this.logger.warn(`[getData] Block ${startingBlock} doesn't exist`);
-      throw new Error(`Block ${startingBlock} doesn't exist`);
-    }
-    data.block = new RskBlock(
-      lastBlock.number,
-      lastBlock.hash,
-      lastBlock.parentHash
-    );
-    for (const transaction of lastBlock.transactions) {
-      if (transaction.to !== bridge.address) {
-        continue;
-      }
-
-      this.logger.trace(`Found a bridge tx ${transaction.hash} with signature ${transaction.input.substring(0, 10)}`);
 
-      if (this.filters.length == 0 ||
-        this.filters.some(f => f.isMethodCall(transaction.input))
-      ) {
-        this.logger.debug(`Tx ${transaction.hash} matches filters`);
-
-        const tx = new RskTransaction();
-        tx.blockHeight = lastBlock.number;
-        tx.blockHash = lastBlock.hash;
-        tx.createdOn = new Date(Number(lastBlock.timestamp) * 1000);
-        tx.hash = transaction.hash;
-        tx.data = transaction.input;
-        const txReceipt = await this.rskNodeService.getTransactionReceipt(tx.hash);
-        tx.logs = <Array<Log>>txReceipt.logs;
-        data.data.push(tx);
+  async process(rskBlock: RskBlock): Promise<void> {
+    this.logger.debug(`[process] Processing rskBlock ${rskBlock.hash}`);
+    for(const transaction of rskBlock.transactions) {
+      for(const subscriber of this.subscribers) {
+        const filters = await subscriber.getFilters();
+        if (filters.length === 0 || filters.some(f => f.isMethodCall(transaction.data))) {
+          this.logger.debug(`[process] Tx ${transaction.hash} matches filters`);
+          const txReceipt = await this.rskNodeService.getTransactionReceipt(transaction.hash);
```

Comment:
I suggest defining this txReceipt above the subscribers loop, in case the same transaction would apply for more than one subscriber.
```suggestion
          if (!txReceipt) {
            txReceipt = await this.rskNodeService.getTransactionReceipt(transaction.hash);
          }
```